### PR TITLE
python3Packages.monai: init at 0.9.0

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Description of changes

Add MONAI, a Pytorch-based library for deep learning in medical imaging.  (Refresh of stale #147538.)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [NA] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [NA] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [NA] (Package updates) Added a release notes entry if the change is major or breaking
  - [NA] (Module updates) Added a release notes entry if the change is significant
  - [NA] (Module addition) Added a release notes entry if adding a new NixOS module
  - [NA] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
You haven't included the `python-packages.nix` addition in the commit.

**Comment by bcdarwin** [comment]
Fixed now, sorry.

File: pkgs/development/python-modules/monai/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "Project-MONAI";
+    repo = "MONAI";
+    rev = "${version}";
+    sha256 = "sha256-HxW9WYxt2a7fS9/1E9DtiH+SCTTJoxYBfgZqskYdcvI=";
+  };
+
+  patchPhase = ''
```
Comment (smancill) [inline pkgs/development/python-modules/monai/default.nix:25]:
```suggestion
  postPatch = ''
```

No need to override the phase.

Comment (bcdarwin) [reply pkgs/development/python-modules/monai/default.nix]: Fixed.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 182263` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.monai</li>
    <li>python39Packages.monai</li>
  </ul>
</details>

**Comment by smancill** [comment]
Import fails in the `nixpkgs-review` shell for Python 3.10:

```
Python 3.10.5 (main, Jul 14 2022, 15:49:56) [Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import monai
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/kfznbn0iifx5f4pfydmnqgvydikjkckp-python3.9-monai-0.9.0/lib/python3.9/site-packages/monai/__init__.py", line 38, in <module>
    from .utils.module import load_submodules  # noqa: E402
  File "/nix/store/kfznbn0iifx5f4pfydmnqgvydikjkckp-python3.9-monai-0.9.0/lib/python3.9/site-packages/monai/utils/__init__.py", line 15, in <module>
    from .deprecate_utils import DeprecatedError, deprecated, deprecated_arg
  File "/nix/store/kfznbn0iifx5f4pfydmnqgvydikjkckp-python3.9-monai-0.9.0/lib/python3.9/site-packages/monai/utils/deprecate_utils.py", line 19, in <module>
    from monai.utils.module import version_leq
  File "/nix/store/kfznbn0iifx5f4pfydmnqgvydikjkckp-python3.9-monai-0.9.0/lib/python3.9/site-packages/monai/utils/module.py", line 26, in <module>
    import torch
  File "/nix/store/f0mgn9j01yxhfh7smkypyfayg7hqd89v-python3.9-pytorch-1.11.0/lib/python3.9/site-packages/torch/__init__.py", line 216, in <module>
    raise ImportError(textwrap.dedent('''
ImportError: Failed to load PyTorch C extensions:
    It appears that PyTorch has loaded the `torch/_C` folder
    of the PyTorch repository rather than the C extensions which
    are expected in the `torch._C` namespace. This can occur when
    using the `install` workflow. e.g.
        $ python setup.py install && python -c "import torch"

    This error can generally be solved using the `develop` workflow
        $ python setup.py develop && python -c "import torch"  # This should succeed
    or by running Python from a different directory.
```

But it works with Python 3.9 (and it builds with both versions)

**Comment by bcdarwin** [comment]
Weird -- it imports fine for me under `nix-shell`.  In your case it appears that `import torch` has failed, possibly because you're somehow picking up `lib/python3.9` libs, which seems wrong?

**Comment by smancill** [comment]
I checked out the branch manually and tried with `nix-shell`, it works with both Python 3.9 and Python 3.10.

Something weird going on with `nixpkgs-review`.

**Review by smancill — APPROVED** [review]

**Comment by trofi** [comment]
Does fetch work for you? hydra seems to fail to get the correct file. Fails for me as well on `master`:

```
$ nix build -f. python3Packages.monai.src
error: hash mismatch in fixed-output derivation '/nix/store/52hh3z4fxv7cqb6nkpm05p4k4sc3550f-source.drv':
         specified: sha256-HxW9WYxt2a7fS9/1E9DtiH+SCTTJoxYBfgZqskYdcvI=
            got:    sha256-Kgi0m3OuLHCIo/gl35Hn2fyOWr2lcwNgswgKeWBGJPQ=
error:
```

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 182263` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.monai</li>
    <li>python39Packages.monai</li>
  </ul>
</details>

**Comment by bcdarwin** [comment]
@trofi did you mean to build #188308?

It does seem like the hash for the 0.9.0 tag has changed.

**Comment by trofi** [comment]
Ah, I did not see newer pull request. Ran against it as well.

**Pull request merged**